site-src/geps/gep-724.md

```diff
@@ -0,0 +1,426 @@
+  // only kind of target resource with "Core" support is Gateway. This API may
+  // be extended in the future to support additional kinds of parent resources
+  // such as Routes.
+  //
+  // +optional
+  // +kubebuilder:validation:MaxItems=16
+  AttachTo []AttachRef `json:"gateways,omitempty"`
+```
+
+And the struct that references:
+```go
+// AttachRef identifies an API object to attach to. The only kind of target
+// resource with "Core" support is Gateway. This API may be extended in the
+// future to support additional kinds of parent resources such as Routes.
+type AttachRef struct {
+  // Group is the group of the referent.
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +kubebuilder:default=networking.x-k8s.io
+  Group string `json:"group"`
+
+  // Kind is kind of the referent.
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  Kind string `json:"kind"`
+
+  // Name is the name of the referent.
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  Name string `json:"name"`
+
+  // SectionName is the name of a section within the target resource. When
+  // unspecified, this targets the entire resource. In the following
+  // resources, SectionName is interpreted as the following:
+  // * Gateway: Listener Name
+  // * Route: Rule Name
+  //
+  // Support: Core
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +optional
+  SectionName string `json:"sectionName,omitempty"`
+
+  // Namespace is the namespace of the referent. When unspecified, the local
+  // namespace is inferred unless targeting a cluster-scoped resource in which
+  // case no namespace is inferred.
+  //
+  // Support: Extended
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +optional
+  Namespace string `json:"namespace,omitempty"`
+}
+```
+
+### Advantages
+
+* Simplifies the API by providing a single way to attach Routes to Gateway.
+* Assigns clear responsibilities to Gateway and Route owners.
+* Consistent with pattern of direct references from Routes to all associated
+  resources.
+* Enables attaching Routes to arbitrary parents, such as custom Gateways, other
+  Routes (to be defined), or Meshes.
+* Prevents accidental exposure of Routes.
+* Easy to understand which Gateways/parents a Route is attached to.
+* Further simplifies path to Route inclusion.
+* Follows pattern of direct reference in one direction with a broader trust
+  reference in the other direction.
+* Aligns with initial user feedback.
+
+### Disadvantages
+
+* Attaching a Route to a named listener with SectionName may be a bit confusing.
+* Does not utilize existing ReferencePolicy mechanism.
+* May be more difficult to understand which Routes are attached to a Gateway.
+* Adding/replacing a Gateway requires changes to Routes.
```

Comment:
This one is a deviation from our previous conversations around the separation of routes and gateways, which is okay.
Any commentary on how to avoid this?

Reply:
Yeah this is one that made me a bit sad. I really liked the ability to seamlessly spin up new Gateways and target the same Routes. The initial feedback we've received from our first UXR sessions suggests that's not actually as valuable as I thought and that app owners would rather have full control on the Gateways their app is attached to.

One note here is that this proposal isn't really adding anything new, it's just taking away other ways to connect Gateways and Routes. I think it makes sense to start with the simplest form possible for v1alpha2 and as requests come in, we can always consider adding back additional concepts such as selectors. I'm just not sure that request is going to be as common as I'd originally thought.